airbyte-integrations/connectors/source-github/source_github/source.py

```diff
@@ -146,8 +146,21 @@ def check_connection(self, logger: AirbyteLogger, config: Mapping[str, Any]) ->
             for stream_slice in repository_stats_stream.stream_slices(sync_mode=SyncMode.full_refresh):
                 next(repository_stats_stream.read_records(sync_mode=SyncMode.full_refresh, stream_slice=stream_slice), None)
             return True, None
+
         except Exception as e:
-            return False, repr(e)
+            message = repr(e)
+            if '404 Client Error: Not Found for url: https://api.github.com/repos/' in message:
```

Comment:
Can we have the `status_code` and `error message` as a separate variables and check for match based on them?
something like:
The code is just an example, check the methods for `e` based on what you have as an exception.
```
error_code = e.response.status_code
error_msg = e.response.json().get("message")

if error_code in [404, ...] and error_msg in ["https://api.github.com/repos/"]:
    ....
```

Reply:
yes, I guess the GitHub API should return normal error responses. We have to check this